File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/contract/SetParkingLotsContract.java
package com.mobile.tandil.javabaseproject.mvp.contract;

public interface SetParkingLotsContract {

    interface SetParkingLotsPresenter {
        void onSubmitButtonPressed();

        void onCancelButtonPressed();
    }

    interface SetParkingLotsView {
        void closeSetParkingLotsDialog();

        String getParkingLots();

        void showParkingLots(int lots);

        void showErrorMessage();
    }
}

File: app/src/main/java/com/mobile/tandil/javabaseproject/mvp/contract/SetParkingLotsContract.kt
package com.mobile.tandil.javabaseproject.mvp.contract

import com.mobile.tandil.javabaseproject.databinding.DialogFragmentSetParkingLotsBinding

interface SetParkingLotsContract {
    interface Presenter {
        fun onSubmitButtonPressed()
        fun onCancelButtonPressed()
        fun getBinding(): DialogFragmentSetParkingLotsBinding
    }

    interface View {
        fun onSubmitBtnClick(onClick: () -> Unit)
        fun onCancelBtnClick(onClick: () -> Unit)
        fun closeSetParkingLotsDialog()
        fun getParkingLots(): String
        fun showParkingLots(lots: Int)
        fun showErrorMessage()
        fun getBinding(): DialogFragmentSetParkingLotsBinding
    }
}
